lib/proxy/launcher/webrick.rb

```diff
@@ -1,73 +1,74 @@
-require 'proxy/log'
-require 'proxy/settings'
-require 'proxy/signal_handler'
-require 'proxy/log_buffer/trace_decorator'
 require 'sd_notify'
 
-CIPHERS = ['ECDHE-RSA-AES128-GCM-SHA256', 'ECDHE-RSA-AES256-GCM-SHA384',
-           'AES128-GCM-SHA256', 'AES256-GCM-SHA384', 'AES128-SHA256',
-           'AES256-SHA256', 'AES128-SHA', 'AES256-SHA'].freeze
-
-module Proxy
-  class Launcher
+module Launcher
+  class Webrick
     include ::Proxy::Log
 
-    attr_reader :settings
+    attr_reader :launcher
 
-    def initialize(settings = SETTINGS)
-      @settings = settings
+    def initialize(launcher)
+      @launcher = launcher
     end
 
-    def pid_path
-      settings.daemon_pid
-    end
+    def launch
+      http_app = build_http_app
+      https_app = build_https_app
+      install_webrick_callback!(http_app, https_app)
 
-    def http_enabled?
-      !settings.http_port.nil?
-    end
+      t1 = Thread.new { webrick_server(https_app, settings.bind_host, settings.https_port).start } unless https_app.nil?
+      t2 = Thread.new { webrick_server(http_app, settings.bind_host, settings.http_port).start } unless http_app.nil?
 
-    def https_enabled?
-      settings.ssl_private_key && settings.ssl_certificate && settings.ssl_ca_file
-    end
+      Proxy::SignalHandler.install_traps
```

Comment:
My comment is actually a leftover from my shutdown investigation, it's not relevant. These traps are only for webrick, maybe we should rename this method to show these are only for Webrick to prevent confusion.

Reply:
You found it out before I could reply. That's exactly the reasoning. Perhaps the SignalHandler should be merged in the Webrick launcher or at least moved to `Proxy::Launcher::Webrick::SignalHandler`.